crates/cargo-contract/src/cmd/verify.rs

```diff
@@ -62,9 +72,100 @@ impl VerifyCommand {
     pub fn run(&self) -> Result<VerificationResult> {
         let manifest_path = ManifestPath::try_from(self.manifest_path.as_ref())?;
         let verbosity: Verbosity = TryFrom::<&VerbosityFlags>::try_from(&self.verbosity)?;
+        if let Some(path) = &self.contract {
+            self.verify_contract(manifest_path, verbosity, path)
+        } else if let Some(path) = &self.wasm {
+            self.verify_wasm(manifest_path, verbosity, path)
+        } else {
+            anyhow::bail!("Either --wasm or --contract must be specified")
+        }
+    }
+
+    /// Verify `.wasm` binary.
+    fn verify_wasm(
+        &self,
+        manifest_path: ManifestPath,
+        verbosity: Verbosity,
+        path: &PathBuf,
+    ) -> Result<VerificationResult> {
+        // 1. Read code hash binary from the path.
+        let file = File::open(path)
+            .context(format!("Failed to open contract binary {}", path.display()))?;
+
+        let mut ref_reader = BufReader::new(file);
+        let mut ref_buffer = Vec::new();
+        ref_reader
+            .read_to_end(&mut ref_buffer)
+            .context(format!("Failed to read contract binary {}", path.display()))?;
```

Comment:
Could be shortened to:

```suggestion
       let ref_buffer = std::fs::read(path)
            .context(format!("Failed to read contract binary {}", path.display()))?;
```